cranelift/codegen/src/isa/x64/inst/mod.rs

```diff
@@ -2542,6 +2572,28 @@ fn x64_get_operands(inst: &mut Inst, collector: &mut impl OperandVisitor) {
             mem.get_operands(collector);
         }
 
+        Inst::LockXadd {
+            operand,
+            mem,
+            dst_old,
+            ..
+        } => {
+            collector.reg_use(operand);
+            collector.reg_reuse_def(dst_old, 0);
+            mem.get_operands(collector);
+        }
+
+        Inst::Xchg {
+            operand,
+            mem,
+            dst_old,
+            ..
+        } => {
+            collector.reg_use(operand);
+            collector.reg_reuse_def(dst_old, 0);
+            mem.get_operands(collector);
+        }
```

Comment:
I believe this is correct but @cfallin do you want to double-check?

Reply:
Looks right to me!